pkg/codegen/schema/loader.go

```diff
@@ -228,3 +242,113 @@ func LoadPackageReference(loader Loader, pkg string, version *semver.Version) (P
 	}
 	return p.Reference(), nil
 }
+
+func (l *pluginLoader) loadSchemaBytes(pkg string, version *semver.Version) ([]byte, *semver.Version, error) {
+	if err := l.ensurePlugin(pkg, version); err != nil {
+		return nil, nil, err
+	}
+
+	pluginInfo, err := l.host.ResolvePlugin(workspace.ResourcePlugin, pkg, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if version == nil {
+		version = pluginInfo.Version
+	}
+
+	if pluginInfo.SchemaPath != "" {
+		schemaBytes, ok := l.loadCachedSchemaBytes(pkg, pluginInfo.SchemaPath, pluginInfo.SchemaTime)
+		if ok {
+			return schemaBytes, nil, nil
+		}
+	}
+
+	schemaBytes, provider, err := l.loadPluginSchemaBytes(pkg, version)
+	if err != nil {
+		return nil, nil, fmt.Errorf("Error loading schema from plugin: %w", err)
+	}
+
+	if pluginInfo.SchemaPath != "" {
+		err = atomic.WriteFile(pluginInfo.SchemaPath, bytes.NewReader(schemaBytes))
+		if err != nil {
+			return nil, nil, fmt.Errorf("Error writing schema from plugin to cache: %w", err)
+		}
+	}
+
+	if version == nil {
+		info, err := provider.GetPluginInfo()
+		if err != nil {
+			// Nonfatal
```

Comment:
I think this error check is wrong. Do you mean
```go
if err == nil {
    version = info.Version
}
```

Right now you assign from `info.Version` even though it could be nil. See [`providerServer.GetPluginInfo`](https://github.com/pulumi/pulumi/blob/f3e430a1da2b665f29d711b9f1bfd9b9c7f240cc/sdk/go/common/resource/plugin/provider_server.go#L138-L144).

FYI: We have `contract.IgnoreError` if we just want to swallow the error.

Reply:
Yep, that's what I meant :)